crates/bevy_render/src/shader/shader.rs

```diff
@@ -121,17 +121,38 @@ impl Shader {
         }
     }
 
-    #[allow(unused_variables)]
+    #[cfg(not(target_arch = "wasm32"))]
     pub fn get_spirv_shader(&self, macros: Option<&[String]>) -> Shader {
         Shader {
-            #[cfg(not(target_arch = "wasm32"))]
             source: ShaderSource::Spirv(self.get_spirv(macros)),
-            #[cfg(target_arch = "wasm32")]
-            source: self.source.clone(),
             stage: self.stage,
         }
     }
 
+    #[cfg(target_arch = "wasm32")]
+    pub fn get_spirv_shader(&self, macros: Option<&[String]>) -> Shader {
```

Comment:
I'd like to future-proof this for an eventual WebGPU backend if possible. I don't want to conflate "wasm" with "webgl"

Reply:
(where i think we would use the other variant of this function)

Reply:
Yes, I also don't like these `target_arch = "wasm32"` checks here. Maybe we should check presence of some `spirv` feature or so?

Reply:
or delegate this to RenderResourceContext too?

Reply:
Yeah that seems reasonable

Reply:
done - shader specialization delegated to RenderResourceContext.get_specialized_shader